aspnet/security/cross-site-scripting.rst

```diff
@@ -1,8 +1,256 @@
+also affects Razor TagHelper and HtmlHelper rendering as it will use the encoders to output your strings.
+
+The reasoning behing this is to protect against unknown or future browser bugs (previous browser bugs have tripped up parsing based on the processing of non-English characters). If your 
+web site makes heavy use of non-Latin characters, such as Chinese, Cyrillic or others this is probably not the behavior you want.
+
+You can customize the encoder safe lists to include Unicode ranges appropriate to your application during startup, in ``ConfigureServices()``.
+
+For example, using the default configuration you might use a Razor HtmlHelper like so;
+
+.. code-block:: html
+
+  <p>This link text is in Chinese: @Html.ActionLink("汉语/漢語", "Index")</p>
+
+When you view the source of the web page you will see it has been rendered as follows, with the Chinese text encoded;
+
+.. code-block:: html
+
+  <p>This link text is in Chinese: <a href="/">&#x6C49;&#x8BED;/&#x6F22;&#x8A9E;</a></p>
+
+To widen the characters treated as safe by the encoder you would insert the following line into the ``ConfigureServices()`` method in ``startup.cs``;
+
+.. code-block:: c#
+
+  services.AddSingleton<HtmlEncoder>(
+    HtmlEncoder.Create(allowedRanges: new[] { UnicodeRanges.BasicLatin, 
+                                              UnicodeRanges.CjkUnifiedIdeographs }));
+
+This example widens the safe list to include the Unicode Range CjkUnifiedIdeographs. The rendered output would now become
+
+.. code-block:: html
+
+    <p>This link text is in Chinese: <a href="/">汉语/漢語</a></p>
+
+Safe list ranges are specified as Unicode code charts, not languages. The `Unicode standard <http://unicode.org/>`_ has a list of `code charts <http://www.unicode.org/charts/index.html>`_ 
+you can use to find the chart containing your characters. Each encoder, Html, JavaScript and Url, must be configured separately.
+
+.. NOTE::  
+  Customization of the safe list only affects encoders sourced via DI. If you directly access an encoder via ``System.Text.Encodings.Web.*Encoder.Default`` then the default,
+  Basic Latin only safelist will be used.
+
+Where encoding should take place?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The general accepted practice is that encoding takes place at the point of output and encoded values should never be stored in a database. Encoding at the point of output 
+allows you to change the use of data, for example, from HTML to a query string value. It also enables you to easily search your data without having to encode values before searching 
+and allows you to take advantage of any changes or bug fixes made to encoders.
+
+Validation as an XSS prevention technique
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Validation can be a useful tool in limiting XSS attacks. For example a simple numeric string, containing only the characters 0-9 will not trigger an XSS attack. Validation becomes
```

Comment:
I'd move the comma
For example, a simple numeric string containing only the characters 0-9 will not trigger an XSS attack.